lib/idx/remediate.ts

```diff
@@ -164,7 +184,20 @@ export async function remediate(
       }
     }
   }
-  
+
+  // check if public method parameters match actions in relatesTo object
+  const relatesToAction = getRelatesToAction(idxResponse, values);
```

Comment:
how about
1. map actions from values, the function does not need to be too general, we can evolve it over time.
2. add action to the preset actions array in line 172, then it can fall into the existing logic to handle actions
3. remove the action from the actions array when the async call is finished